astropy/nddata/ccddata.py

```diff
@@ -416,7 +416,11 @@ def _generate_wcs_and_update_header(hdr):
     wcs_header = wcs.to_header(relax=True)
     for k in wcs_header:
         if k not in _KEEP_THESE_KEYWORDS_IN_HEADER:
-            del new_hdr[k]
+            try:
+                del new_hdr[k]
+            except KeyError:
+                # Keyword does not exist in the original header
+                pass
```

Comment:
So this deletes any keywords that aren't 'JD-OBS',  'MJD-OBS', or 'DATE-OBS'? Why remove WCSAXES for example?